### ISSUE TITLE
Crash on app termination in mbgl::platform::Settings::get

### ISSUE THREAD

**Issue by parrots** [mapbox/mapbox-maps-ios#984]
## Environment

- Xcode version: 13.2
- iOS version: iOS 14 and 15
- Devices affected: iPhone and iPad
- Maps SDK Version: 10.2

## Observed behavior

Crash when app is being terminated:

```
Crashed: com.apple.main-thread
0  libsystem_pthread.dylib        0x7744 pthread_mutex_lock + 102
1  libc++.1.dylib                 0x3caf4 std::__1::mutex::lock() + 16
2  MapboxCoreMaps                 0x1caee8 mbgl::platform::Settings::get(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const + 2314 (memory:2314)
3  MapboxCoreMaps                 0xc67a8 mbgl::Tracer::getInstance() + 1423 (string:1423)
4  MapboxCoreMaps                 0x423578 mbgl::internal::ScopedTraceSectionData<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::~ScopedTraceSectionData() + 55 (tracer.hpp:55)
5  MapboxCoreMaps                 0x423d14 std::__1::__function::__func<mbgl::style::Style::Impl::loadURL(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::$_1, std::__1::allocator<mbgl::style::Style::Impl::loadURL(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::$_1>, void (mbgl::Response)>::destroy_deallocate() + 3212 (memory:3212)
6  MapboxCoreMaps                 0xe6fec mbgl::FileSourceRequest::Impl::~Impl() + 7 (file_source_request.cpp:7)
7  MapboxCoreMaps                 0xe6c44 std::__1::__function::__func<mbgl::FileSourceRequest::~FileSourceRequest()::$_0, std::__1::allocator<mbgl::FileSourceRequest::~FileSourceRequest()::$_0>, void ()>::destroy() + 3212 (memory:3212)
8  MapboxCommon                   0x3e578 _hidden#969_ + 2536
9  MapboxCoreMaps                 0xc5c94 std::__1::__deque_base<std::__1::unique_ptr<mapbox::common::Task, std::__1::default_delete<mapbox::common::Task> >, std::__1::allocator<std::__1::unique_ptr<mapbox::common::Task, std::__1::default_delete<mapbox::common::Task> > > >::clear() + 325 (deque:325)
10 MapboxCoreMaps                 0xc5b38 mbgl::util::RunLoop::~RunLoop() + 1187 (deque:1187)
11 libsystem_c.dylib              0x58270 __cxa_finalize_ranges + 408
12 libsystem_c.dylib              0x585c4 exit + 28
13 UIKitCore                      0xbd62f0 -[UIApplication terminateWithSuccess] + 502
14 UIKitCore                      0x2228cc -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 128
15 UIKitCore                      0x2224fc -[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:] + 220
16 UIKitCore                      0xbcbdc0 -[UIApplication workspaceShouldExit:withTransitionContext:] + 212
17 FrontBoardServices             0x69a8 -[FBSUIApplicationWorkspaceShim workspaceShouldExit:withTransitionContext:] + 88
18 FrontBoardServices             0x353d4 __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke_2 + 80
19 FrontBoardServices             0x196a4 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
20 FrontBoardServices             0x3536c __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke + 132
21 libdispatch.dylib              0x481c _dispatch_client_callout + 20
22 libdispatch.dylib              0x830c _dispatch_block_invoke_direct + 268
23 FrontBoardServices             0x5dfa0 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
24 FrontBoardServices             0x5dc30 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 448
25 FrontBoardServices             0x5e184 -[FBSSerialQueue _performNextFromRunLoopSource] + 32
26 CoreFoundation                 0xa3990 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
27 CoreFoundation                 0xa388c __CFRunLoopDoSource0 + 208
28 CoreFoundation                 0xa2bfc __CFRunLoopDoSources0 + 376
29 CoreFoundation                 0x9cb70 __CFRunLoopRun + 820
30 CoreFoundation                 0x9c308 CFRunLoopRunSpecific + 600
31 GraphicsServices               0x3734 GSEventRunModal + 164
32 UIKitCore                      0xbca75c -[UIApplication _run] + 1072
33 UIKitCore                      0xbcffcc UIApplicationMain + 168
34 Slopes                         0x9df84 main + 25 (main.swift:25)
35 libdyld.dylib                  0x1cf8 start + 4
```

<img width="715" alt="Screen Shot 2021-12-30 at 11 57 31" src="https://user-images.githubusercontent.com/171163/147780547-56386ca6-39f4-4113-824d-0d811f2a5493.png">

**Comment by ZiZasaurus** [comment]
Ticketed upstream for GL-Native

**Comment by parrots** [comment]
<img width="730" alt="Screen Shot 2022-02-22 at 11 12 08" src="https://user-images.githubusercontent.com/171163/155193422-1ece0de5-bdaf-427f-bd70-9e1927245a2a.png">

I am seeing this greatly reduced, if not fixed entirely, since pushing 10.3 out into production last week.

**Comment by ZiZasaurus** [comment]
This has been resolved in 10.3.
